src/posts/introducing-pint-xarray/index.md

```diff
@@ -0,0 +1,292 @@
+
+[This wrapping is currently necessary](https://xarray.pydata.org/en/stable/user-guide/duckarrays.html#missing-features) for any operation which needs to be aware of the units of a dimension coordinate of the dataarray, or any xarray operation which relies on an external library (such as calling `scipy` in `.integrate`).
+
+## CF-compliant units for geosciences with cf-xarray
+
+Different fields tend to have different niche conventions about how certain units are defined.
+By default, pint doesn't understand all the unusual units and conventions we use in geosciences.
+But [pint is customisable](https://pint.readthedocs.io/en/stable/defining.html), and with the help of [cf-xarray](https://github.com/xarray-contrib/cf-xarray) we can teach it about these geoscience-specific units.
+
+If we `import cf_xarray.units` (before `import pint_xarray`) then we can `quantify` example data from a pangeo climate data store :
+
+```python
+import xarray as xr
+import cf_xarray.units  # must come before import pint_xarray
+import pint_xarray
+
+ds = xr.open_dataset('gs://cmip6/CMIP6/CMIP/NCAR/CESM2-FV2/historical/r2i1p1f1/Amon/sfcWind/gn/v20200226/', engine='zarr')
+ds = ds.pint.quantify()
+
+squared_wind = ds['sfcWind'] ** 2
+squared_wind.pint.units
+```
+
+```
+Out: <Unit('meter ** 2 / second ** 2')>
+```
+
+Here (thanks to `cf_xarray`) pint has successfully interpreted the CF-style units `'m s-1'`, then automatically changed them when we squared the wind speed.
+
+## Plotting
+
+We can complete our real-world example by plotting the data in its new units
+
+```python
+import cartopy.crs as ccrs
+import matplotlib.pyplot as plt
+
+p = squared_wind.isel(time="2014-01").plot(
+    subplot_kws=dict(projection=ccrs.Orthographic(-80, 35), facecolor="gray"),
+    transform=ccrs.PlateCarree(),
+)
+p.axes.set_global()
+p.axes.coastlines()
+plt.show()
+```
+
+![cartopy plot of a quantified dataset](/posts/introducing-pint-xarray/squared_wind.png)
+
+where `xarray.plot` has detected the pint units automatically.
+
+## Conclusion
+
+Please have a go! You will need xarray (v0.20+) and pint-xarray.
+
+Please also tell us about any bugs you find, or documentation suggestions you have on the [xarray](https://github.com/pydata/xarray/issues) or [pint-xarray issue trackers](https://github.com/xarray-contrib/pint-xarray/issues).
+If you have usage questions you can raise them there, on the [xarray discussions page](https://github.com/pydata/xarray/discussions), or on the [pangeo discourse forum](https://discourse.pangeo.io/).
+
+The work here to allow xarray to wrap pint objects is part of a [broader effort to generalise xarray](http://xarray.pydata.org/en/stable/roadmap.html#flexible-arrays) to handle a wide variety of data types (so-called "duck array wrapping").
+Along with the incoming [support for flexible indexes](http://xarray.pydata.org/en/stable/roadmap.html#flexible-indexes), we are excited for all the new features that this will enable for xarray users!
```

Comment:
```suggestion
The work here to allow Xarray to wrap Pint objects is part of a [broader effort to generalize Xarray](http://xarray.pydata.org/en/stable/roadmap.html#flexible-arrays) to handle a wide variety of data types (so-called "duck array wrapping").
Along with the incoming [support for flexible indexes](http://xarray.pydata.org/en/stable/roadmap.html#flexible-indexes), we are excited for all the new features that this will enable for Xarray users!
```